Rbac reduce guard clause

kbrock opened the pull request:
/cc @matthewd @gtanzillo minor, but couldn't resist

imtayadeway commented on app/models/rbac.rb:
```diff
@@ -186,8 +186,7 @@ def self.compute_total_count(klass, scope, extra_target_ids, conditions, include
     klass.reflections.each do |association, reflection|
-      next unless association == association_to_match
-      return reflection
+      return reflection if association == association_to_match
```
Could this be further simplified with `detect` instead of `each`/`return`?

imtayadeway commented on app/models/rbac.rb:
```diff
@@ -226,14 +225,12 @@ def self.find_targets_filtered_by_ids(klass, scope, find_options, u_filtered_ids
   def self.get_belongsto_filter_object_ids(klass, filter)
-    return nil if filter.nil? || filter.empty?
-    return nil unless BELONGSTO_FILTER_CLASSES.include?(safe_base_class(klass).name)
+    return nil if filter.nil? || filter.empty? || !BELONGSTO_FILTER_CLASSES.include?(safe_base_class(klass).name)
```
Doesn't `empty?` also cover the `nil` case?

kbrock commented:
@imtayadeway Guess I didn't reduce enough

kbrock commented:
I pulled out the other guard clause, I may want access to the objects.

imtayadeway commented:
@kbrock nice! :+1:

matthewd commented on app/models/rbac.rb:
```diff
@@ -185,11 +185,7 @@ def self.compute_total_count(klass, scope, extra_target_ids, conditions, include
-      return reflection
-    end
-    nil
+    klass.reflections.detect { |association, reflection| association == association_to_match }
```
Doesn't this change the behaviour?

kbrock replied:
ugh.

matthewd replied:
The last one returns `[association, reflection]`

kbrock replied:
and this is why I didn't change this the first time around :(

also of note, this is called with a symbol, but all the keys in reflections are strings.

curious why we're not using `klass.reflections[association_to_match.to_s]`

imtayadeway replied:
:point_up: my bad

kbrock commented:
Looks like there were bugs in this code.

Revisiting with more specs

kbrock commented:
Fixed bug around association not properly coming back.
Looks like we use `association_id` behind the scenes, so this bug is hidden with most of the models I checked.

I also added documentation around the various rbac parameters. I'm hoping to get these PRs flowing so I can get some of my larger changes into there.

chessbyte commented:
@gtanzillo please review

gtanzillo commented:
:+1: LGTM
